### PR TITLE
added priority list

### PR DESCRIPTION
**Pull request by muskankhedia**
Added a priority list to check for the input and perform the action according to the priority, i.e, if a user has an input ```google weather Bhubaneswar Odisha bing```, it will return the weather conditions ignoring the Google and Bing search as the weather is higher in the priority list.
The priority list in descending order is set as:
<i>images, image, video, videos, watch, youtube, symptoms, medicine, weather, meaning, google, yahoo, bing, search</i>

Some other tests of this priority list are shown in the [video.](https://drive.google.com/file/d/1PjVl4vas-_yONmev2IpdicP-XvdIMXnN/view)

### CODE REVIEW

File: service/controllers/messages.go [hunk 1]
```diff
@@ -98,66 +98,104 @@ func routes(routeObject response, w http.ResponseWriter) {
...
+	// 			strArr=strings.Split(strArr0," ")
+	// 			remainingString=strings.Join(stringDifference(strArr,strArrDiff)," ")
+	// 			fmt.Println(remainingString)
+	// 		}
+	// 	} else {
+	// 			strArr = strings.Split(matchPars, " ")
+	// 			strArrDiff = strings.Split(message, " ")
+	// 			remainingString=strings.Join(stringDifference(strArr,strArrDiff)," ")
+	// 			fmt.Println(remainingString)
+	// 		}
```
Comment (Harkishen-Singh) [inline service/controllers/messages.go]: please remove these comments. I have reviewed the code and looks cool!

File: service/controllers/messages.go [hunk 2]
```diff
@@ -98,66 +98,104 @@ func routes(routeObject response, w http.ResponseWriter) {
...
+	// }
+
+	message := routeObject.message
+	messageArr := strings.Fields(message)
+	var a []string
+	priority := []string{"images", "image", "video", "videos", "watch", "youtube", "symptoms", "medicine", "weather", 
+	"meaning", "google", "yahoo", "bing", "search"}
```
Comment (Harkishen-Singh) [inline service/controllers/messages.go]: give a ***tabs*** indent for the *meaning ...* line

File: service/controllers/messages.go [hunk 3]
```diff
@@ -98,66 +98,104 @@ func routes(routeObject response, w http.ResponseWriter) {
...
+	for i := 0; i < len(messageArr); i++ {
+		for j := 0; j < len(priority); j++ {
```
Comment (Harkishen-Singh) [inline service/controllers/messages.go]: when using a loop for traversing the array lists, always use forEach functionality! apply this to both the above loops.
a change of only 4 lines would do this work!

File: service/controllers/messages.go [hunk 4]
```diff
@@ -98,66 +98,104 @@ func routes(routeObject response, w http.ResponseWriter) {
...
+		matchPars = sort[0]
+		remainingString = strings.Join(messageArr[:]," ")
+		messageArr = append([]string{matchPars}, messageArr...)
+		fmt.Println("Remaining String:", remainingString)
+		fmt.Println("matchPars: ", matchPars)
+	} else {
+		remainingString = strings.Join(messageArr[:], " ")
+		fmt.Println("messageArr:", messageArr)
+		fmt.Println("Remaining String:", remainingString)
+		fmt.Println("matchPars: ", matchPars)
```
Comment (Harkishen-Singh) [inline service/controllers/messages.go]: please remove ***print*** statements if the code is working according to your plan

File: service/controllers/messages.go [hunk 5]
```diff
@@ -390,6 +428,26 @@ func routes(routeObject response, w http.ResponseWriter) {
 
 }
 
+func customSort(arr1 []string, arr2 []string, m,n int) []string{
```
Comment (Harkishen-Singh) [inline service/controllers/messages.go]: add comments before the ***customSort()*** as a help to the contributors

File: service/controllers/messages.go (hunk 1 above)
Comment (muskankhedia) [reply service/controllers/messages.go]: Yaa...Thank you.

File: service/controllers/messages.go (hunk 3 above)
Comment (muskankhedia) [reply service/controllers/messages.go]: I think, we can add the forEach functionality only for the inner loop, as the value of the outer loop has to be changed explicitly, which cannot be changed while using forEach. Pls, suggest if I am thinking in the wrong way.

Comment (Harkishen-Singh) [reply service/controllers/messages.go]: sure. please go ahead!

Comment (muskankhedia) [reply service/controllers/messages.go]: Pls have a review, I have made the changes accordingly.

**Pull request merged**